src/main/native/io_sqooba_traildb_TrailDBNative.cpp

```diff
@@ -482,35 +479,44 @@ JNIEXPORT jint JNICALL Java_io_sqooba_traildb_TrailDBNative_tdbCursorNext
 	tdb_cursor *cursor = (tdb_cursor*) env->GetDirectBufferAddress(jcursor);
 
 	// Call lib.
-	const tdb_event *event = tdb_cursor_next(cursor);
+	const tdb_event *event;
+	event = tdb_cursor_next(cursor);
 
 	// Check if there is no more events.
 	if(!event) {
 		return -1;
 	}
 
-	// Get struct elements.
-	uint64_t timestamp = event->timestamp;
-	uint64_t num_items = event->num_items;
-	const tdb_item *items_ptr = event->items;
-
+	// Store field in TrailDBEvent.
+	env->SetLongField(jevent, JFID_traildbEvent_timestamp, event->timestamp);
+	env->SetLongField(jevent, JFID_traildbEvent_numItems, event->num_items);
+	env->SetLongField(jevent, JFID_traildbEvent_items, (long)event->items);
 
-  	jlongArray newArray = env->NewLongArray(num_items);
-    jlong *narr = env->GetLongArrayElements(newArray, NULL);
+	return 0;
+}
 
-	unsigned int i;
-    for (i = 0; i < num_items; i++) {
-        narr[i] = items_ptr[i];
-    }
+JNIEXPORT jstring JNICALL Java_io_sqooba_traildb_TrailDBNative_eventGetItemValue
```

Comment:
This method is returning String only. Is that what we want or just a temporary limitation ?
If it is what we want I would make the method name more explicit (e.g. eventGetItemStringValue)

Reply:
An item value is always a String and can not be something else.
Since it is always the case, I don't find it is necessary to say it in the method name. Moreover the name will then differ too much from the C method name in my opinion.